datadog_lambda/wrapper.py

```diff
@@ -40,19 +47,29 @@ class _LambdaDecorator(object):
 
     def __init__(self, func):
         self.func = func
-        self.flush_to_log = os.environ.get('DD_FLUSH_TO_LOG', '').lower() == 'true'
-        self.logs_injection = os.environ.get('DD_LOGS_INJECTION', '').lower() == 'true'
+        self.flush_to_log = os.environ.get("DD_FLUSH_TO_LOG", "").lower() == "true"
+        self.logs_injection = os.environ.get("DD_LOGS_INJECTION", "").lower() == "true"
 
         # Inject trace correlation ids to logs
         if self.logs_injection:
             inject_correlation_ids()
 
         # Patch HTTP clients to propagate Datadog trace context
         patch_all()
-        logger.debug('datadog_lambda_wrapper initialized')
+        logger.debug("datadog_lambda_wrapper initialized")
 
     def _before(self, event, context):
+        global cold_start_request_id
```

Comment:
I'm pretty sure `context.aws_request_id` stays the same for retried requests, which is gonna cause incorrect cold start?  Also if you are not using the actual request id, how about?

```python
# In a `cold_start.py` or `utils.py`?
initialized = False
cold_start = True

def set_cold_start():
    global initialized
    global cold_start
    cold_start = (initialized == False)
    initialized = True

def is_cold_start():
    global cold_start
    return cold_start

# In wrapper.py (here)
from utils import set_cold_start
def _before(self, event, context):
    set_cold_start()

# use is_cold_start in, say, metric.py
from utils import is_cold_start
is_cold_start()
```

Reply:
Might be better to prefix the globals with `_` to mark them as "private".